src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -324,7 +324,7 @@ private string GenerateForTypeWithUnknownConverter(TypeGenerationSpec typeMetada
                             }}
                             else
                             {{
-                                throw new {InvalidOperationExceptionTypeRef}($""The converter '{{converter.GetType()}}' is not compatible with the type '{{typeToConvert}}'."");
```

Comment:
With this, we are localizing run-time messages in the generated code, even though we don't localize other run-time exceptions thrown from the runtime AFAIK. Do other generators do this?

My previous [thoughts ](https://github.com/dotnet/runtime/issues/58292#issuecomment-910491645)on this we that we shouldn't localize run-time exceptions since we don't do this elsewhere; we should only localize messages shown within VS as errors\warnings. If \ when we decide to localize messages in the runtime, then I think that would be the best time to do this for generated code. Localizing runtime messages may not always be desired -- it does make it harder to search for the message, and increases download size.

Also we generate the message with the current VS locale. Is that what developers want? They may want the message of the current locale that is running and not the locale that it was developed with.